core/src/main/resources/META-INF/resources/primefaces-extensions/keynote/widget.js

```diff
@@ -27,16 +27,48 @@ PrimeFaces.widget.ExtKeynote = PrimeFaces.widget.BaseWidget.extend({
             margin: cfg.margin || 0.04,
             minScale: cfg.minScale || 0.2,
             maxScale: cfg.maxScale || 2.0,
-            
-            plugins: [ RevealMarkdown ]
+
+            plugins: [RevealMarkdown]
         };
+
+        if (this.options.slideNumber === "false") {
+            this.options.slideNumber = false;
+        }
         
-        if (this.options.slideNumber === "false") this.options.slideNumber = false;
+        this.keynote = new Reveal(document.querySelector(this.jqId), this.options);
+        this.startKeynote();
+    },
 
-        let deck = new Reveal(document.querySelector(this.jqId), this.options);
-        deck.initialize();
+    /**
+     * Starts the keynote.
+     */
+    startKeynote: function () {
+        var $this = this;
 
-    }
+        $this.keynote.initialize().then(() => {
+            $this.keynote.on('slidetransitionend', event => {
```

Comment:
Why do you map this to "end" to only detect the last slide? Why not simply have a slidechanged event.

Reply:
I removed "end", added `slidechanged` and `slidetransitionend`

Reply:
I see you used their style, but I would prefer `slideChange` and `slideTransitionEnd`. See for example https://primefaces.github.io/primefaces/11_0_0/#/components/datatable?id=ajax-behavior-events